crates/common/src/lib.rs

```diff
@@ -267,6 +267,10 @@ impl BlockNumber {
     pub fn is_zero(&self) -> bool {
         self == &Self::GENESIS
     }
+
+    pub fn checked_sub(&self, rhs: u64) -> Option<Self> {
+        self.0.checked_sub(rhs).map(Self)
+    }
```

Comment:
Think this is identical to the `parent` method. This is probably a more conventional name though so maybe replace `parent` with this?

Reply:
Ah right apologies.